source/common/http/filter_manager.h

```diff
@@ -1075,7 +1081,12 @@ class FilterManager : public ScopeTrackedObject,
       std::make_shared<Network::Socket::Options>();
 
   FilterChainFactory& filter_chain_factory_;
+  // Used to track local reply state
+  // TODO(esmet): We could combine the local reply reference and data into one pointer
+  // member that stays nullptr when no local reply config is present to save space in
+  // the 'off' case. But, for now, we spend two pointers of space in all cases.
   const LocalReply::LocalReply& local_reply_;
+  Utility::LocalReplyDataPtr local_reply_data_;
```

Comment:
This is still a TODO since I think it will be a requirement to not regress memory usage in the case where Envoy is not configured to use any local reply rewrites.